### ISSUE TITLE
Control flow cross-references

### ISSUE THREAD

**Issue by toolCHAINZ** [Vector35/binaryninja-api#1044]
Asked about this in slack, but figured I'd open an issue just in case:

In an architecture I’m writing, there are some indirect jumps using a jump table. The way the code I’m looking at calculates the address to jump to is to calculate the in-table offset for the jump, then add that to the offset of the start of the table from the data segment, and then add in the address of the data segment. MLIL looks like this:

```python
# reg2 has a possible value set of [0, 4: 1]
reg1 = reg2 << 1                           #in-table offset
reg1 = 0x1000 + reg1                       #jump-table entry offset from segment start
reg1 = [seg_start + (reg1 & 0x3fff)].w     #load from absolute address (masking out parts of the register not to be used for addresses)
jump(another_pointer + reg1)
```

Binary Ninja is able to figure out possible values that reg1 is pointing to and use them for jump table detection (creating jumptable arrays in the process, and correctly finding call the branches in the function). However, I don't see a cross reference from the function to the jumptable. Instead, a cross-reference appears at `seg_start`. Since Binary Ninja is doing the work of figuring out what offsets from seg start are actually accessed, I would really like it if those addresses could also have cross references from this code.

Not really knowing how any of this works, my guess is that this isn't currently happening because the actual absolute addresses only ever exist as the result of dataflow, and never as an `il.const` entry.

**Comment by plafosse** [comment]
Our intention here is to add cross-references to things that effect control flow.

**Comment by plafosse** [comment]
FWIW You can now manually create cross-references in the UI now.

**Comment by psifertex** [comment]
We believe that cross-references to the jump table itself should be showing up based on some recent changes. That said, lookup tables do not have references which we're tracking under #1856. I'm going to close this on the assumption that the existing jump table cross-reference improvement has resolved that particular issue, but feel free to re-open if that's not the case!

**Comment by bpotchik** [comment]
This has been fixed as of `3.0.3295-dev`
